content/docs/implementation-notes.md

```diff
@@ -9,175 +9,174 @@ redirect_from:
   );
 }
 
-// This function takes a React element (e.g. <App />)
-// and returns a DOM or Native node representing the mounted tree.
+// Fungsi ini mengambil elemen React (misalnya <App />)
+// dan mengembalikan DOM atau *node* Native yang mewakili diagram yang dipasang.
 function mount(element) {
   var type = element.type;
   var props = element.props;
 
-  // We will determine the rendered element
-  // by either running the type as function
-  // or creating an instance and calling render().
+  // Kita akan menentukan elemen yang di *render*
+  // dengan menjalankan tipe sebagai fungsi
+  // atau membuat *instance* dan memanggil render().
   var renderedElement;
   if (isClass(type)) {
-    // Component class
+    // Komponen kelas
     var publicInstance = new type(props);
-    // Set the props
+    // Atur *props*
     publicInstance.props = props;
-    // Call the lifecycle if necessary
+    // Panggil *lifecycle* jika diperlukan
     if (publicInstance.componentWillMount) {
       publicInstance.componentWillMount();
     }
-    // Get the rendered element by calling render()
+    // Dapatkan elemen yang di *render* dengan memanggil render()
     renderedElement = publicInstance.render();
   } else {
-    // Component function
+    // Komponen fungsi
     renderedElement = type(props);
   }
 
-  // This process is recursive because a component may
-  // return an element with a type of another component.
+  // Proses ini rekursif karena komponen mungkin
+  // mengembalikan elemen dengan tipe komponen lain.
   return mount(renderedElement);
 
-  // Note: this implementation is incomplete and recurses infinitely!
-  // It only handles elements like <App /> or <Button />.
-  // It doesn't handle elements like <div /> or <p /> yet.
+  // Catatan: implementasi ini tidak lengkap dan berulang-ulang!
+  // Ini hanya menangani elemen seperti <App /> atau <Button />.
+  // Ini tidak menangani elemen seperti <div /> atau <p /> belum.
 }
 
 var rootEl = document.getElementById('root');
 var node = mount(<App />);
 rootEl.appendChild(node);
 ```
 
->**Note:**
+>**Catatan:**
 >
->This really *is* a pseudo-code. It isn't similar to the real implementation. It will also cause a stack overflow because we haven't discussed when to stop the recursion.
+>Ini benar-benar *pseudo-code*. Itu tidak mirip dengan implementasi nyata. Itu juga akan menyebabkan *stack overflow* karena kita belum membahas kapan harus menghentikan rekursi.
 
-Let's recap a few key ideas in the example above:
+Mari kita rekap beberapa ide utama dalam contoh di atas:
 
-* React elements are plain objects representing the component type (e.g. `App`) and the props.
-* User-defined components (e.g. `App`) can be classes or functions but they all "render to" elements.
-* "Mounting" is a recursive process that creates a DOM or Native tree given the top-level React element (e.g. `<App />`).
+* Element React adalah objek polos yang mewakili jenis komponen(misalnya. `App`) dan *props*.
```

Comment:
```suggestion
* Elemen React adalah objek polos yang mewakili jenis komponen (mis. `App`) dan *props*.
```
- element -> elemen
- "e.g." could be translated to "mis".

Reply:
Hoho sorry for typo. OK Thanks